### ISSUE TITLE
Docs on return type for IReactorUNIX, IReactorArbitrary missing

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4838]
|<img alt="msmith's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| msmith reported|
|-|-|
|Trac ID|trac#1609|
|Type|enhancement|
|Created|2006-03-28 14:27:35Z|

There's no docs for the return type of the connect/listen methods in IReactorUNIX, IReactorArbitrary . Patch to be attached.

Attachments:

* [reactorinterfaces.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ead/ead98e0e365d07598a35ba682210564193928571/55e57f45c13b90814e3d9e4aca7269269d7c645e.patch) (1321 bytes) - added by msmith on 2006-03-28 14:28:05Z - Patch to add return type docs.

<details><summary>Searchable metadata</summary>

```
trac-id__1609 1609
type__enhancement enhancement
reporter__msmith msmith
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1143556055000000 1143556055000000
changetime__1149643457000000 1149643457000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun|
|-|-|

I started to apply this but it looks like there's a lot of garbage in interfaces.py.

  * "satisfies" doesn't mean anything.  There are two choices when talking about interfaces - implements and provides: the correct choice in the case of connectXYZ and listenXYZ is "provides".

 * IReactorSSL is missing a @returns too

 * IReactorTCP has more information than any of the other interfaces

These are not problems with the patch, just problems which the patch brought to light :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

Patch applied and many more changes made in [branches/internet-interface-1609](branches/internet-interface-1609) - please review.

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub set owner to @glyph|
|-|-|

I really love that the docstrings are getting improved, and I have gone through each of the docstrings, and have noted a few things that aren't consistent.

@return and @returns are synonyms, but '@return' is used more often. Please use it in the patch.

getDestination, getHost and getPeer needs a @return.

writeToChild does not specify parameters or return value in the docstring. Why is this a new function? This is not mentioned in the ticket notes. I don't object to it being added, but please note in the ticket why it's been introduced.

the log.msg() in the end of the diff does this anti-pattern:
```
    log.msg("foo " +
            "bar " +
            "baz")
```
which can be spelled:
```
    log.msg("foo "
            "bar "
            "baz")
```
and the string concatenation will be precomputed in the scanner, instead of done at run time.

Please correct these things, and when you have, merge this branch.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [17103]) Merge internet-interfaces-1609

Author: exarkun
Reviewer: Jerub
Fixes [#4838](https://github.com/twisted/twisted/issues/4838)

Change the language in docstrings in twisted/internet/interfaces.py to be more 
consistent and use terms which actually exist.  Also add several missing pieces
of information relating to the process interfaces.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
